pkg/alertmanager/multitenant.go

```diff
@@ -825,13 +825,22 @@ func (am *MultitenantAlertmanager) setConfig(cfg alertspb.AlertConfigDesc) error
 	var userAmConfig *amconfig.Config
 	var err error
 	var hasTemplateChanges bool
+	var userTemplateDir = filepath.Join(am.getTenantDirectory(cfg.User), templatesDir)
+	var filesToRemove = make(map[string]string)
```

Comment:
Is there a reason to store the filename and the path? It feels like we could just store the path. Having both implies there's an important distinction here (if there is, and I'm missing it, than that's fine, just maybe add a comment why).

```suggestion
	var pathsToRemove = make(map[string]struct{})
```